packages/snap-preact-components/src/components/Atoms/RecommendationResultTracker/RecommendationResultTracker.tsx

```diff
@@ -0,0 +1,64 @@
+/** @jsx jsx */
+import { h } from 'preact';
+import { jsx, css } from '@emotion/react';
+import { useRef } from 'preact/hooks';
+import { observer } from 'mobx-react-lite';
+import { Theme, useTheme } from '../../../providers';
+import { useIntersection } from '../../../hooks';
+import type { RecommendationController } from '@searchspring/snap-controller';
+import { ComponentProps, StylingCSS } from '../../../types';
+import type { Product } from '@searchspring/snap-store-mobx';
+import classnames from 'classnames';
+
+const CSS = {
+	RecommendationResultTracker: () => css({}),
+};
+
+export const RecommendationResultTracker = observer((properties: RecommendationResultTrackerProps): JSX.Element => {
+	const globalTheme: Theme = useTheme();
+
+	const props: RecommendationResultTrackerProps = {
+		// default props
+		// global theme
+		...globalTheme?.components?.RecommendationResultTracker,
+		// props
+		...properties,
+		...properties.theme?.components?.RecommendationResultTracker,
+	};
+
+	const { children, result, controller, className, disableStyles, style } = props;
+
+	const resultRef = useRef(null);
+	const resultInViewport = useIntersection(resultRef, '0px');
+
+	if (resultInViewport) {
+		if (!controller.events.impression) {
+			controller.track.impression();
```

Comment:
Should we be doing this?

I'm starting to think maybe this would encourage improper setups, or at the very least would lead to reports being partially correct which would make it hard to identify an invalid setup.

I have more to say on this in the main comment for the PR.